lib/hooks/core/hook-http.js

```diff
@@ -35,6 +35,7 @@ function requestWrap(request) {
 
     var localOptions = (typeof options === 'string') ?
       url.parse(options) : options;
+    localOptions.headers = localOptions.headers || {};
```

Comment:
An alternative here would be to do:
```javascript
options = (..) ? url.parse(options) : options
```
We are trying to be careful to avoid changing the caller's options object, but if that is really our intent; the code at present is not rigorous enough. This statement adds a header property on the original options object if the original was not a string.

So: either make it more rigorous (and add a comment) or relent.